torch/csrc/dynamo/python_compiled_autograd.cpp

```diff
@@ -369,7 +402,9 @@ variable_list compiled_autograd(
   }
 
   // TODO(jansel): some dynamic sizes seem to be ints not symints
-  if (!cache->check_dynamic_sizes(compiler_call)) {
+  // TODO(voz): Replace with a general cache check fn
+  if (!cache->check_dynamic_sizes(compiler_call) ||
+      !cache->check_keep_graph(compiler_call)) {
```

Comment:
I think instead of doing this, we should just encode keep_graph in the cache key.

We could do something right after:
https://github.com/pytorch/pytorch/blob/4bbe41bc05f48ec5dc61559887ced9a983019630/torch/csrc/dynamo/python_compiled_autograd.cpp#L361C22-L361C40
like:
```
if(keep_graph) {
   cache = cache->lookup(... fixed key to mark usage of keep_graph ...)
}
```

Reply:
Given the caching discussion, I believe that this diff can be deleted entirely